internal/controller/promotion/argocd.go

```diff
@@ -76,6 +79,14 @@ func (a *argoCDMechanism) Promote(
 		return promo.Status.WithPhase(kargoapi.PromotionPhaseSucceeded), newFreight, nil
 	}
 
+	if a.argocdClient == nil {
+		return promo.Status.WithPhase(kargoapi.PromotionPhaseFailed), newFreight,
+			errors.New(
+				"Argo CD integration is disabled on this controller; cannot perform " +
+					"promotion",
+			)
+	}
```

Comment:
It isn't just the Application reconciler that we cannot start when Argo CD is disabled. We also cannot carry out Argo CD-based promotion mechanisms.